### PR TITLE
[Maintenance] drop support for node versions no longer security supported. Now >= 14. 

### LINKED ISSUE #296: Drop support for node versions that are no longer security supported.
Chatted with @pkanal and we think we should only keep and test for Node 14, 16, 18. The others are no longer security supported. 

Testing end-of-life versions of node also blocks dependency updates (see: https://github.com/honeycombio/libhoney-js/pull/293)

Places to change this:

- [x] our CI tests "12.22", "14.17", "15.14", "16.2", "17.0", "18.0"
- [x] our package.json we also bump it to node Version 14
- [x] see what happens ✨

### PR DESCRIPTION
**Pull request by emilyashley**
## Which problem is this PR solving?

- closes #296 
- unblocks dependency updates #293 #304 

## Short description of the changes

- CircleCI to only test Node 14, 16, 18 (current security supported node versions)
- Update readme and package.json and package lock to Node >=14
- I noticed the env variable in the example app  `HONEY_WRITE_KEY` was different than what most of our examples used, `HONEYCOMB_API_KEY`, so I changed it to be consistent :)

### CODE REVIEW

**Review by MikeGoldsmith — APPROVED** [review]
Looks good, thanks @emilyashley 👍🏻

File: examples/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ To build and run these examples:
 1. `cd $example-dir`
 1. `npm install ../.. # this installs the libhoney you built above`
 1. `npm install`
-1. `HONEY_WRITE_KEY=YOUR_WRITE_KEY npm start`
+1. `HONEYCOMB_API_KEY=YOUR_WRITE_KEY npm start`
```
Comment (JamieDanielson) [inline examples/README.md:7]: Since this PR is for the major version bump in dropping support for old Node, I think we should move these example changes to a different PR to keep the focus on the major change.

Comment (emilyashley) [reply examples/README.md]: Cool, reverted and I'll make a new PR with those.

Comment (emilyashley) [reply examples/README.md]: @JamieDanielson https://github.com/honeycombio/libhoney-js/pull/309 ✨
